quixstreams/platforms/quix/config.py

```diff
@@ -531,84 +522,38 @@ def _set_workspace_cert(self, timeout: Optional[float] = None) -> str:
-        :param timeout: response timeout (seconds); Default 30
-
-        :return: a dict of confluent-kafka-python client settings (see librdkafka
-        config for more details)
+        Get the full client config required to authenticate a confluent-kafka
+        client to a Quix platform broker/workspace as a ConnectionConfig
         """
-        self.get_workspace_info(known_workspace_topic=known_topic, timeout=timeout)
-        cfg_out = {}
-        for quix_param_name, rdkafka_param_name in _QUIX_PARAMS_NAMES_MAP.items():
-            # Map broker config received from Quix to librdkafka format
-            param_value = self.quix_broker_config[quix_param_name]
-
-            # Also map values of "security.protocol" and "sasl.mechanisms" from Quix
-            # to librdkafka format
-            if rdkafka_param_name == "security.protocol":
-                param_value = _QUIX_SECURITY_PROTOCOL_MAP[param_value]
-            elif rdkafka_param_name == "sasl.mechanisms":
-                param_value = _QUIX_SASL_MECHANISM_MAP[param_value]
-            cfg_out[rdkafka_param_name] = param_value
-
-        # Specify SSL certificate if it's provided for the broker
-        ssl_cert_path = self._set_workspace_cert()
-        if ssl_cert_path is not None:
-            cfg_out["ssl.ca.location"] = ssl_cert_path
+        librdkafka_dict = self.api.get_librdkafka_connection_config(self.workspace_id)
+        if (cert := librdkafka_dict.pop("ssl.ca.cert", None)) is not None:
+            cert = base64.b64decode(cert)
+            librdkafka_dict["ssl.ca.location"] = self._set_workspace_cert(cert)
+            self._librdkafka_connection_config = ConnectionConfig.from_librdkafka_dict(
+                librdkafka_dict
+            )
 
-        # Set the connection idle timeout and metadata max age to be less than
-        # Azure's default 4 minutes.
-        # Azure LB kills the inbound TCP connections after 4 mins and these settings
-        # help to handle that.
-        # More about this issue:
-        # - https://github.com/confluentinc/librdkafka/issues/3109
-        # - https://learn.microsoft.com/en-us/azure/event-hubs/apache-kafka-configurations#producer-and-consumer-configurations-1
-        # These values can be overwritten on the Application level by passing
-        # `extra_consumer_config` or `extra_producer_config` parameters.
-        cfg_out["connections.max.idle.ms"] = QUIX_CONNECTIONS_MAX_IDLE_MS
-        cfg_out["metadata.max.age.ms"] = QUIX_METADATA_MAX_AGE_MS
-        self._confluent_broker_config = cfg_out
-        return self._confluent_broker_config
-
-    def get_confluent_client_configs(
-        self,
-        topics: list,
-        consumer_group_id: Optional[str] = None,
-        timeout: Optional[float] = None,
-    ) -> Tuple[dict, List[str], Optional[str]]:
+    def get_application_config(
+        self, consumer_group_id: str
+    ) -> Tuple[ConnectionConfig, dict, Optional[str]]:
```

Comment:
I think we could return here some object with named properties instead, e.g. some dataclass `QuixApplicationConfig`.
It would make the result more readable.